052-cruise-control-api-users.md

```diff
@@ -0,0 +1,204 @@
+
+For adding Cruise Control API users, we could update the Cruise Control spec with an `apiUsers` section.
+In this `apiUsers` section, developers could provide a list of API users by name and security role.
+
+The schema could look something like this:
+
+```yaml=
+apiVersion: kafka.strimzi.io/v1beta2
+kind: Kafka
+metadata:
+  ...
+spec:
+  cruiseControl:
+    apiUsers:
+      - name: user
+        role: USER (3)
+     ...
+```
+
+When defined like above, Strimzi would generate a Secret containing a password for the specified API user.
+
+In addition to the above, we could allow developers to supply a custom password per specified API user via a Secret like this:
+
+```yaml=
+apiVersion: kafka.strimzi.io/v1beta2
+kind: Kafka
+metadata:
+  ...
+spec:
+  cruiseControl:
+    apiUsers:
+      - name: user
+        role: USER (3)
+        password:
+          valueFrom:
+            secretKeyRef:
+              name: my-secret (1)
+              key: my-password (2)
+     ...
+```
+
+(1) The name of the secret containing the predefined password.
+
+(2) The key for the password stored inside the secret.
+
+(3) By default Cruise Control defines three roles: `VIEWER`, `USER` and `ADMIN`. 
+For more details checkout the [Cruise Control Wiki](https://github.com/linkedin/cruise-control/wiki/Security#authorization)
+```
+VIEWER role: has access to the most lightweight kafka_cluster_state, user_tasks and review_board endpoints.
+USER role: has access to all the GET endpoints except bootstrap and train.
+ADMIN role: has access to all endpoints.
+```
+
+Strimzi would then aggregate the Cruise Control API auth credentials into a centralized configuration, then store it in a secret where it would be mounted and used by Cruise Control.
+
+### How API user management is handled elsewhere
+
+Cruise Control is not the only Kafka component which has API users.
+For consitency across the code base, we should also consider the following components when designing our solution:
+
+* Kafka users
+* KafkaConnect users
+
+#### Kafka users
+
+In Strimzi, Kafka users are managed via custom resources.
+
+```yaml=
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaUser
+metadata:
+  name: my-user-1
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  authentication:
+    type: tls
+  authorization:
+    type: simple
+    acls:
+      ...
+```
+These resources are watched and managed by the the Strimzi User Operator, which creates Kafka users based on the descriptions provided by the resources.
+
+For more details, checkout the Strimzi documentation [here](https://strimzi.io/docs/operators/latest/deploying#proc-configuring-kafka-user-str)
+
+#### KafkaConnect users
+
+At this time, Strimzi doesn't offer any method of creating or configuring KafkaConnect API users.
+That being said, the direction we take with creating and managing Cruise Control API users could be used as a precedent for its design in the future.
+
+## Alternatives
```

Comment:
What does this mean? Are you proposing to implement all of these in parallel? Or did you reject these? I think this should have some more clarity.

Reply:
No no, I just wasn't sure which direction to take, but given the feedback, I have moved all alternatives to the rejected alternatives section